### DISCUSSION TITLE
how can I remove extra spaces from inline/multiline comment using eslint ecosystem?

### DISCUSSION THREAD

**Discussion by cakkypamucm** [eslint/eslint#17183] [Q&A] [answered]
Now I have code comment:
`/* eslint-disable-next-line               import/no-unresolved */`

And I want to transform it to (after eslint work):
`/* eslint-disable-next-line import/no-unresolved */`

[no-multi-spaces](https://eslint.org/docs/latest/rules/no-multi-spaces) doesn't work for me.
And after I saw [tests](https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-multi-spaces.js) I'm not sure if `no-multi-spaces` should work in my case.

So, my question is - how can I remove extra spaces from inline/multiline comment using eslint ecosystem? (`eslint` embedded rules or `eslint` special plugins)

Maybe related with [issue](https://github.com/eslint/eslint/issues/8917)

**Comment by sam3k** [comment]
Hi @cakkypamucm, it seems like this [is allowed](https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-multi-spaces.js#L64-L65) for multiline comments. Most likely this is the case because of the [other issue](https://github.com/eslint/eslint/issues/8917) you've linked to where this rule would break JSDocs.

Thinking about this a bit more, having this rule inside a comment itself doesn't make a lot of sense and the `no-multi-space` rule only handles space checks outside the comment itself:

```javascript
var foo = 'hello';     // extra spaces before comment <-- linter will show error 
var boo = 'world'; // extra    spaces   inside    comment <-- linter will (and probably should not) show error as this is a comment
```

[Here](https://eslint.org/play/#eyJ0ZXh0IjoiLyogaGVsbG8gICAgIHdvcmxkICovXG4vLyBoZWxsbyAgICB3b3JsZFxuXG52YXIgZm9vID0gJ2hlbGxvJzsgICAgIC8vIGV4dHJhIHNwYWNlcyBiZWZvcmUgY29tbWVudCA8LS0gbGludGVyIHdpbGwgc2hvdyBlcnJvciBcbnZhciBib28gPSAnd29ybGQnOyAvLyBleHRyYSAgICBzcGFjZXMgICBpbnNpZGUgICAgY29tbWVudCA8LS0gbGludGVyIHdpbGwgKGFuZCBwcm9iYWJseSBzaG91bGQgbm90KSBzaG93IGVycm9yIGFzIHRoaXMgaXMgYSBjb21tZW50Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby1tdWx0aS1zcGFjZXMiOlsiZXJyb3IiLHsiaWdub3JlRU9MQ29tbWVudHMiOmZhbHNlfV19LCJlbnYiOnt9fX0=) is a sandbox. This makes perfect sense to me as we should assume comments will be used in many ways; including JSDocs, really lengthy paragraphs, etc.

**Comment by cakkypamucm** [comment]
@sam3k , of course, _in general_ we should assume comments will be used in many ways.
But in my _specific_ case (and fully controlled of me) I want to remove extra spaces from inline/multiline comment using eslint ecosystem  (`eslint` embedded rules or `eslint` special plugins).
Does such way exist?

Of course, if exceptions will appear in my my _specific_ case, I'll disable them manually using comment `eslint-disable`

**Reply by sam3k** [reply, marked as answer]
@cakkypamucm totally understand what you mean. As far as I can tell, `no-multi-space` was never meant to be used inside a comment. This is one of those cases where it might make sense to make a custom plugin to handle these scenarios. Something like `no-multi-space-in-lint-comment` or something like that. Then parse comments containing ESLint syntax and then do the check.

**Reply by cakkypamucm** [reply]
@sam3k, thank you for your informative and quick feedback!
